dartagnan/src/main/java/com/dat3m/dartagnan/parsers/program/utils/ProgramBuilder.java

```diff
@@ -202,10 +202,7 @@ public MemoryObject getOrNewMemoryObject(String name, int size) {
             mem.setName(name);
             if (program.getFormat() == LITMUS) {
                 // Litmus code always initializes memory
-                final Expression zero = expressions.makeZero(types.getArchType());
-                for (int offset = 0; offset < size; offset++) {
-                    mem.setInitialValue(offset, zero);
-                }
+                mem.setInitialValue(0, expressions.makeZero(types.getIntegerType(8 * size)));
             }
```

Comment:
I don't understand why we need this change? Isn't this the responsible of all the fails in the CI that Thomas mentioned were related to init events?

Reply:
No, I think this change is because before we have never considered the correct type for initial memory.
A 1-byte sized memory object could hold a 8-byte sized initial value, which is just not correct, but didn't result in problems before.

Reply:
This change was necessary.  The former snippet produced overlapping initializations, which would always result in worst-case byte-wise Tearing.  A compromise would be to produce initial values for every eighth byte, as below.
```java
final Expression zero = expressions.makeZero(types.getArchType());
final int archSize =  types.getMemorySizeInBytes(types.getArchType());
for (int offset = 0; offset + archSize <= size; offset += archSize) {
    mem.setInitialValue(offset, zero);
}
if (size % archSize != 0) {
    final IntegerType remType = types.getIntegerType(size % archSize);
    mem.setInitialValue(archSize * (size / archSize), expressions.makeZero(remType));
}
```